### PR TITLE
Add initial Playwright tests

### PR DESCRIPTION
**Pull request by kachkaev**
This PR follows #434 and adds a few simple tests to the repo. They:

- ensure presence of some key elements of the codebase
- ensure navigation between pages generally works (using top menu)
- [Vercel only] ensure `/api/me` returns HTTP 403 instead of 500 – simple DB health check

The diff also contains some setup enhancement (see comments). More tests and setup improvements will be added in follow-up PRs.

### CODE REVIEW

File: .github/workflows/site-deployment.yml [hunk 1]
```diff
@@ -4,7 +4,7 @@ on:
   deployment_status:
   schedule:
     - cron: "*/10 * * * *" ## Every 10 minutes
-    - cron: "0 4 * * *" ## Every night (UTC time)
+    - cron: "0 4 * * *" ## Around 4am every night (UTC time)
```
Comment (kachkaev) [inline .github/workflows/site-deployment.yml:7]: https://github.com/blockprotocol/blockprotocol/pull/434#discussion_r922150283

File: site/playwright.config.ts [hunk 2]
```diff
@@ -5,14 +5,21 @@ const ci = process.env.CI === "true";
...
+      name: "smoke",
+      retries: 0,
+      testMatch: "**/{smoke,universal}/**",
+    },
   ],
   reporter: [
     [ci ? "github" : "list"],
     ["html", { open: !ci ? "on-failure" : "never" }],
   ],
-  retries: 0,
```
Comment (kachkaev) [inline site/playwright.config.ts:15]: I’ve decided to vary the number of retries depending on the project. Inspired by https://playwright.dev/docs/test-advanced#configuration-object

File: .github/workflows/site-deployment.yml [hunk 3]
```diff
@@ -58,3 +58,10 @@ jobs:
         run: yarn workspace @blockprotocol/site playwright test --project=smoke
         env:
           PLAYWRIGHT_TEST_BASE_URL: ${{ github.event.deployment.environment != 'production' && github.event.deployment_status.target_url || 'https://blockprotocol.org' }}
+
+      - name: Upload artifact playwright-report
+        if: ${{ success() || failure() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: site-smoke-tests-playwright-report
+          path: site/playwright-report
```
Comment (kachkaev) [inline .github/workflows/site-deployment.yml:67]: Setup improvement: this artifact helps debugging failed smoke tests if there are any

File: site/README.md [hunk 4]
```diff
@@ -1,11 +1,9 @@
 # blockprotocol.org website
 
-The public-facing [blockprotocol.org](https://blockprotocol.org) website serves the following endpoints:
+This folder contains the code for [blockprotocol.org](https://blockprotocol.org), including:
 
-- `/gallery` shows the catalog of available blocks
-- `/api/blocks` returns a JSON response w/ a list of available blocks
-- `/blocks/<organisation>/<blockname>` is the CDN base-URL of an individual block (e.g. `/blocks/@hash/code`)
-- `/partners` provides a temporary signup form to collect pre-release registrations of interest from potential adopters
```
Comment (kachkaev) [inline site/README.md:8]: Side improvement: remove stale docs

File: site/tests/universal/page-header-navigation.test.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    ),
+  ).toBeVisible();
+
+  await page.locator("header >> text=Documentation").click();
+  await expect(page).toHaveURL("/docs");
+  await expect(page.locator('h1:has-text("Introduction")')).toBeVisible();
+
+  await page.locator("header >> text=Log In").click();
+  await expect(page).toHaveURL("/docs");
+  await expect(page.locator("text=Sign in to theBlock Protocol")).toBeVisible();
```
Comment (nonparibus) [inline site/tests/universal/page-header-navigation.test.ts:21]: Looks odd here that there's no space between "the" and "Block" 😅

File: site/tests/universal/page-header-navigation.test.ts [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  await expect(page).toHaveURL("/docs");
+  await expect(page.locator('h1:has-text("Introduction")')).toBeVisible();
+
+  await page.locator("header >> text=Sign Up").click();
+  await expect(page).toHaveURL("/signup");
+  await expect(
+    page.locator("text=Create your Block Protocol account"),
+  ).toBeVisible();
+
+  // TODO: Add alt to BP logo, ensure that the logo is not clickable from /
```
Comment (teenoh) [inline site/tests/universal/page-header-navigation.test.ts:33]: I think it's okay for the logo to be clickable on all routes. It seems like a common behaviour

**Review by teenoh — APPROVED** [review]

File: site/tests/universal/page-header-navigation.test.ts (hunk 6 above)
Comment (kachkaev) [reply site/tests/universal/page-header-navigation.test.ts:33]: Clicking on a logo is a noop on the home page, so not sure it is needed. I don't have a strong opinion either way 🙂

**Pull request merged**